### PR TITLE
Add metrics for admission attempts count and duration

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Add metrics for admission attempts count and duration, needed to track that the scheduler is doing work.

Add role and rolebinding for prometheus to be able to list the services in the kueue-system namespace.

image

#### Which issue(s) this PR fixes:

Part of #199 (4)

#### Special notes for your reviewer:

This doesn't add the prometheus ServiceMonitor, Role or RoleBinding by default, as users might have their own monitoring system, or they might not want to setup any prometheus at all.

In a follow up, I'll document how users can setup prometheus.

### CODE REVIEW

**Comment by alculquicondor** [comment]
/hold
to prevent premature merge

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			Name:      "admission_attempts_total",
+			Help:      "Number of attempts to admit pods, by result. `success` means that at least one workload was admitted, `inadmissible` means that no workload was admitted.",
+		}, []string{"result"},
+	)
+
+	admissionAttemptLatency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Subsystem: subsystemName,
+			Name:      "admission_attempt_duration_seconds",
+			Help:      "Latency of an admission attempt",
```
Comment (ahg-g) [inline pkg/metrics/metrics.go:48]:
```suggestion
			Help:      "Latency of an admission attempt broken down by result",
```

File: config/default/manager_auth_proxy_patch.yaml [hunk 2]
```diff
@@ -20,6 +20,3 @@ spec:
         - containerPort: 8443
           protocol: TCP
           name: https
-      - name: manager
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply config/default/manager_auth_proxy_patch.yaml:23]: It initially had flags related to metrics, but they are all in the component config now. There is no point of keeping it in the patch.

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	SuccessAdmissionResult      AdmissionResult = "success"
+	InadmissibleAdmissionResult AdmissionResult = "inadmissible"
+)
+
+var (
+	admissionAttempts = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: subsystemName,
+			Name:      "admission_attempts_total",
+			Help:      "Number of attempts to admit pods, by result. `success` means that at least one workload was admitted, `inadmissible` means that no workload was admitted.",
```
[the remark this answers is not in the document]
Comment (ahg-g) [reply pkg/metrics/metrics.go]: sg, we should a gauge metric tracking the number of workloads broken down by pending vs admitted and by ClusterQueue

File: config/prometheus/role.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: prometheus-k8s
+  namespace: system
```
Comment (cmssczy) [inline config/prometheus/role.yaml:5]: Why do we create `Role` in `system` namespace? What is `system` namespace for?

Comment (kerthcet) [reply config/prometheus/role.yaml:5]: well, this is where kustomize works, refer to https://github.com/kubernetes-sigs/kueue/blob/a4703d33fe40045629340b8cf5de4bbcf1d17340/config/default/kustomization.yaml#L4-L9

Comment (cmssczy) [reply config/prometheus/role.yaml:5]: Got it. Thanks

**Pull request merged**